Functions to manage checkboxes and their labels and colors

GerMalaz opened the pull request:
* Definition of Colortext/Colorbold/colortext/colorbold[2] (text, color): Like Regtext/Boldtext/regtext/boldtext with an extra parameter, color

* Definition of ltrcheckbox/rtlcheckbox[4] (checked, label, position, color): Defines a text node with the label to the left of the checkbox node. Position for ltrcheckbox is of the label node. Position for rtlcheckbox is of the checkbox node.

    Example:
`    \checkbox{fo2c} at (135, 197) (fo2c) {};
    \node [t, left, xshift=0.5mm] at (fo2c.west) {\tBoldtext{2\,\c\ }};`
replaced by `\ltrcheckbox{fo2c}{2\bc}{fo1c.east}{black}`

    This could be simplified using a checkbox node with a label, couldn't make that
        to look decent.
        
        
  Assumptions:
            The label is to the left of the checkbox.
            the checkbox and its label have the same color.

GerMalaz committed:
Functions to manage checkboxes and their labels and colors

* Definition of Colortext/Colorbold/colortext/colorbold[2] (text, color):
    Like Regtext/Boldtext/regtext/boldtext with an extra parameter, color

* Definition of ltrcheckbox/rtlcheckbox[4] (checked, label, position, color):
    Defines a text node with the label to the left of the checkbox node.
    Position for ltrcheckbox is of the label node.
    Position for rtlcheckbox is of the checkbox node.

    Example:
`    \checkbox{fo2c} at (135, 197) (fo2c) {};
    \node [t, left, xshift=0.5mm] at (fo2c.west) {\tBoldtext{2\,\c\ }};`
replaced by `\ltrcheckbox{fo2c}{2\bc}{fo1c.east}{black}`

    This could be simplified using a checkbox node with a label, couldn't make that
        to look decent.

GerMalaz commented:
Thinking about generalizing this to create blocks of nodes consisting of:

[ what is used?, (optional?) label, optional checkbox, optional text, optional extra style for individual elements, optional extra style for whole block ] 

Additionally, or orthogonally: A text field could be a block itself of two nodes, a text node and a node with a line below it, optionally drawn [via style definition].

GerMalaz committed:
Merge branch 'mycroftw:master' into master

mycroftw commented:
I like the idea.  However:

I don't think we can get out from *some* hardpositioned (x,y, rather than node.position); I haven't looked to see if that is still possible (knowing LaTeX, it is).

I don't like specific colours in general; yeah, I know, "reg/red/blue" but that's what I was handed from GB.  In general, it should be abstracted: "reg/alert/announce" (and yeah, getting good abbreviations for those is also awful).  If we are going to do a drastic change (and this is the prelude to a drastic change, I'm sure) I'd want the colours to be abstracted away (even though we will *never* change from red/blue/black, we *could*).

Also, would prefer \lcheckbox (or \leftcheckbox) and \r (or \right) rather than RTL or LTR.  I read RTL in the context of gnittamrof egaugnal txet.  not "where's the box in relation to the text".

But yes, this would make things simpler.

GerMalaz commented:
> I like the idea. However:
>
I did some of this in the two_pages_for_gerardo branch in my fork. Converted some of it but stopped short of doing the whole card, and has been a while since

```
\NewDocumentCommand{\ltrblock}{mmmmmmmo}{
    % mandatory: name, position, elements, category,
    % mandatory: labelParameter: labelContent@labelStyle@
    %                            (optional)labelLetterspace
    % mandatory: rangeParameter: rangeLowContent@rangeHighContent@rangeStyle
    % mandatory: textParameter: textContent@textStyle@
    %                           (optional)textLetterspace
    % optional:  fitstyle
```
code inside maybe hairy, some new dependencies, etc.
```
\RequirePackage{bitset}
\RequirePackage{substr}
\RequirePackage{relsize}
\RequirePackage{listofitems}
\RequirePackage{lmodern}
```
code duplication between ltrblock and rtlblock, maybe direction can be another parameter. 
 
> I don't think we can get out from _some_ hardpositioned (x,y, rather than node.position); I haven't looked to see if that is still possible (knowing LaTeX, it is).
> 
Agreed, the position parameter above is either x,y or node.position.
At least some (actually the majority, as of now) is absolute.

> I don't like specific colours in general; yeah, I know, "reg/red/blue" but that's what I was handed from GB. In general, it should be abstracted: "reg/alert/announce" (and yeah, getting good abbreviations for those is also awful). If we are going to do a drastic change (and this is the prelude to a drastic change, I'm sure) I'd want the colours to be abstracted away (even though we will _never_ change from red/blue/black, we _could_).

Done. The fourth parameter, which I named category, can be (so far): regular, alert, announce, regularNoGuideline. Used tikzset to define the related styles. 

> 
> Also, would prefer \lcheckbox (or \leftcheckbox) and \r (or \right) rather than RTL or LTR. I read RTL in the context of gnittamrof egaugnal txet. not "where's the box in relation to the text".

No issues renaming them.
 
> 
> But yes, this would make things simpler.

A few (simple) examples:
```
    \ltrblock{genapproachlabel}{104, 205.5}{1000}{regular}
        {General Approach@bold}
        {}{}
    \rtlblock{genapproach}{203, 205.5}{0001}{regular}
        {}{}
        {\generalapproach @text width=440, centered, bold}
    \ltrblock{mhb}{104, 201}{1000}{regular}
        {Min Expected HCP when Balanced:@italicsLabel}
        {}{}
    \ltrblock{mhbo}{mhb.east}{1001}{regular}
        {Opening}
        {}
        {\minHCPopen @text width=190}
    \rtlblock{mhbr}{203, 201}{1001}{regular}
        {Responding}
        {}
        {\minHCPresp @text width=70, centered}
```
For elements:
```
\newcommand\labelIndex{3}
\newcommand\checkboxIndex{2}
\newcommand\rangeIndex{1}
\newcommand\textIndex{0}
```
And one more thing: I moved the geometry definition to the sty file.

GerMalaz commented:
Redone/refactored, better organized, in #96.
